numpy/core/numeric.py

```diff
@@ -1527,15 +1527,12 @@ def rollaxis(a, axis, start=0):
 
     """
     n = a.ndim
-    if axis < 0:
-        axis += n
+    axis = resolve_axis(axis, n)
     if start < 0:
         start += n
```

Comment:
Yes, I have been caught by that too, realising only slowly that I had to put `array.ndim` if I wanted something moved to the last axis. I think this is partially why we now have `moveaxis` -- in any case, I think it is fine to leave this one untouched!